app/v1/controllers/requestParsers/RetrieveUKPropertyBISSRequestDataParser.scala

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright 2019 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package v1.controllers.requestParsers
+
+import javax.inject.Inject
+import uk.gov.hmrc.domain.Nino
+import v1.controllers.requestParsers.validators.RetrieveUKPropertyBISSValidator
+import v1.models.des.IncomeSourceType
+import v1.models.domain.TypeOfBusiness
+import v1.models.requestData.{DesTaxYear, RetrieveUKPropertyBISSRawData, RetrieveUKPropertyBISSRequest}
+
+class RetrieveUKPropertyBISSRequestDataParser @Inject()(val validator: RetrieveUKPropertyBISSValidator)
+  extends RequestParser[RetrieveUKPropertyBISSRawData, RetrieveUKPropertyBISSRequest] {
+
+  override protected def requestFor(data: RetrieveUKPropertyBISSRawData): RetrieveUKPropertyBISSRequest = {
+    RetrieveUKPropertyBISSRequest(Nino(data.nino),
+      data.taxYear.map(DesTaxYear.fromMtd),
```

Comment:
Naren has added a set of functions which can be found in `DateUtils` which populate this parameter differently.  We want to default this to the current tax year here if no date is supplied.   Check out `RetrieveSelfEmploymentBISSRequestParser`

Reply:
(and the associated tests)

Reply:
This should now be defaulting?